### PR TITLE
RD-5605 Add a description to secrets properties

### PR DESCRIPTION
**Pull request by PatrycjaKubiczek**
## Description

RD-5605
- added label to Form Field
- added test
- mockup description since we don't have a description set yet

PS. could we create another ticket to fix the placement of Popup? I think it should be above (?) icon. Currently is a bit off to the right. 

## Screenshots / Videos

Zrzut ekranu 2022-09-02 131101

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3526/

## Documentation

n/a

### CODE REVIEW

File: app/components/GettingStartedModal/steps/SecretsStep.tsx [hunk 1]
```diff
@@ -38,11 +43,10 @@ const SecretsStep = ({ selectedEnvironment, typedSecrets, onChange }: Props) =>
                     onChange(secretInputs);
                 };
                 return (
-                    <Form.Field key={name}>
+                    <Form.Field key={name} label={label} help={hasSecretDescrtiption ? secretDescription : undefined}>
```
Comment (qooban) [inline app/components/GettingStartedModal/steps/SecretsStep.tsx:46]: I think we shouldn't add "Secret description" to all secrets. We just need to have the framework ready to support new property.

Regarding change in that line, I'm wondering if we change L41 to:
```
            {selectedEnvironment.secrets.map(({ name, label, type, description }) => {
```
could we have here just:
```suggestion
                    <Form.Field key={name} label={label} help={description}>
```
? 🤔

File: app/components/GettingStartedModal/steps/SecretsStep.tsx [hunk 2]
```diff
@@ -25,6 +28,8 @@ const SecretsStep = ({ selectedEnvironment, typedSecrets, onChange }: Props) =>
     );
 
     const [secretInputs, setSecretInputs, resetSecretInputs] = useInputs(typedSecrets || defaultSecretInputs);
+    const secretDescription = t('secretsDescription');
+    const hasSecretDescrtiption = secretDescription;
```
Comment (qooban) [inline app/components/GettingStartedModal/steps/SecretsStep.tsx:32]: Do we need it? 

1. We don't have secret descriptions yet defined, and the goal of RD-5605 was to add support for `description` property. No need to define default value for all secrets, I think.

2. `hasSecretDescription` variable name suggests that this is a boolean containing information if certain secret has description defined. The value is a string "Secret description".

**Review by qooban — COMMENTED** [review]
On the visual side - I think all is fine.

On the implementation side - I think we don't need to add any default help descriptions right now, we should just add support for new property.

Reg.
> PS. could we create another ticket to fix the placement of Popup? I think it should be above (?) icon. Currently is a bit off to the right.

Yes, you're right. It looks like generic problem with `PopupHelp` component. Feel free to create a ticket for that.

File: app/components/GettingStartedModal/steps/SecretsStep.tsx (hunk 1 above)
Comment (PatrycjaKubiczek) [reply app/components/GettingStartedModal/steps/SecretsStep.tsx]: https://github.com/cloudify-cosmo/cloudify-stage/pull/2160/commits/53662860fd59f312c20c62992405bb3b63c24b0f applied here :)

File: app/components/GettingStartedModal/steps/SecretsStep.tsx (hunk 2 above)
Comment (PatrycjaKubiczek) [reply app/components/GettingStartedModal/steps/SecretsStep.tsx]: thanks for the clarification, I think I misunderstood the ticket's goal a bit :) implemented changes

**Review by qooban — COMMENTED** [review]
Looks good 🙂 

Just tests need to be fixed (wrote some ideas on Slack).

**Comment by PatrycjaKubiczek** [comment]
@qooban so I decided to remove the test for now, as we would need to mock up the new secret. I think it'll be very easy to test after adding description - and as it is planned to add description anyway in the future, I'll leave it at it is right now, if you don't mind :)

**Review by qooban — APPROVED** [review]

**Pull request merged**